stko/optimizers/collapser.py

```diff
@@ -228,60 +231,571 @@ def optimize(self, mol):
+        mol,
+        atom_ids,
+        start_vector,
+        target_vector,
+        axis
+    ):
+
+        return None
+
+    def _test_move(self, curr_pot, new_pot):
+
+        if new_pot < curr_pot:
+            return True
+        else:
+            exp_term = np.exp(-self._beta*(new_pot-curr_pot))
+            rand_number = random.random()
+
+            if exp_term > rand_number:
+                return True
+            else:
+                return False
+
+    def _output_top_lines(self):
+
+        string = (
+            '====================================================\n'
+            '                Collapser optimisation              \n'
+            '                ----------------------              \n'
+            '                                                    \n'
+            f' step size = {self._step_size} \n'
+            f' target bond length = {self._target_bond_length} \n'
+            f' num. steps = {self._num_steps} \n'
+            f' bond epsilon = {self._bond_epsilon} \n'
+            f' nonbond epsilon = {self._nonbond_epsilon} \n'
+            f' nonbond sigma = {self._nonbond_sigma} \n'
+            f' nonbond mu = {self._nonbond_mu} \n'
+            f' beta = {self._beta} \n'
+            '====================================================\n\n'
+        )
+
+        return string
+
+    def _plot_progess(self, steps, maxds, spots, npots, output_dir):
+
+        fig, ax = plt.subplots(figsize=(8, 5))
+        ax.plot(steps, maxds, c='k', lw=2)
+        # Set number of ticks for x-axis
+        ax.tick_params(axis='both', which='major', labelsize=16)
+        ax.set_xlabel('step', fontsize=16)
+        ax.set_ylabel('max long bond length [angstrom]', fontsize=16)
+        ax.axhline(y=self._target_bond_length, c='r')
+        fig.tight_layout()
+        fig.savefig(
+            os.path.join(output_dir, f'maxd_vs_step.pdf'),
+            dpi=360,
+            bbox_inches='tight'
+        )
+        plt.close()
+        # Plot energy vs timestep.
+        fig, ax = plt.subplots(figsize=(8, 5))
+        ax.plot(steps, spots, c='k', lw=2, label='total')
+        ax.plot(steps, npots, c='r', lw=2, label='non bonded')
+        # Set number of ticks for x-axis
+        ax.tick_params(axis='both', which='major', labelsize=16)
+        ax.set_xlabel('step', fontsize=16)
+        ax.set_ylabel('potential', fontsize=16)
+        ax.legend(fontsize=16)
+        fig.tight_layout()
+        fig.savefig(
+            os.path.join(output_dir, f'pot_vs_step.pdf'),
+            dpi=360,
+            bbox_inches='tight'
+        )
+        plt.close()
+
+    def optimize(self, mol):
+        """
+        Optimize `mol`.
+
+        Parameters
+        ----------
+        mol : :class:`stk.ConstructedMolecule`
+            The molecule to be optimized.
+
+        Returns
+        -------
+        mol : :class:`stk.ConstructedMolecule`
+            The optimized molecule.
+
+        """
+
+        random.seed()
```

Comment:
This will mean the code is never reproducable as the seed is given by the system time. I'm against.

Reply:
It is a monte carlo algorithm, it should use a new random seed each time.

Reply:
I don't understand why it has to use a different seed each time? Surely you can seed the random number generator once and then keep sampling random numbers using that particular seed.